.doom.d/config.org
#+title: Jacob's Doom Emacs Config
#+author: Jacob Hilker
#+description: Jacob's Doom Emacs config.
#+startup: overview
* Personal Info
Here is where I setup personal info.
#+begin_src emacs-lisp
(setq user-full-name "Jacob Hilker"
      user-mail-address "jacob.hilker2@gmail.com")
#+end_src

* UI Tweaks
** Theme
I really like gruvbox.
#+begin_src emacs-lisp
  (setq doom-theme 'doom-gruvbox)
#+end_src

** Fonts
I personally like Iosevka as a monospace font. In addition, I need to find a good font for variable widths.
#+begin_src emacs-lisp
(setq doom-font (font-spec :family "Iosevka Nerd Font" :size 14)
      doom-variable-pitch-font (font-spec :family "Iosevka Aile" :size 14))

(after! doom-themes
  (setq
   doom-themes-enable-bold t
   doom-themes-enable-italic t))

#+end_src

*** Italics for Comments
#+begin_src emacs-lisp
(custom-set-faces!
  '(font-lock-comment-face :slant italic))
#+end_src

** Org Mode Faces
#+begin_src emacs-lisp
(after! org
  (set-face-attribute 'org-level-1 nil :foreground "#83a598")
  (set-face-attribute 'org-level-2 nil :foreground "#d3869b")
  (set-face-attribute 'org-level-3 nil :foreground "#fabd2f")
  (set-face-attribute 'org-level-4 nil :foreground "#fb4934")
  (set-face-attribute 'org-level-5 nil :foreground "#83a598")
  (set-face-attribute 'org-level-6 nil :foreground "#d3869b")
  (set-face-attribute 'org-level-7 nil :foreground "#fabd2f")
  (set-face-attribute 'org-level-8 nil :foreground "#fb4934"))
#+end_src

* Org Mode Config

#+begin_src emacs-lisp
(after! org
(setq
      org-directory "~/Dropbox/org"
      org-agenda-files '("~/Dropbox/org/agenda.org" "~/Dropbox/org/projects.org" "~/Dropbox/org/contacts.org" "~/Dropbox/org/todo.org")
      org-todo-keywords '((sequence "TODO(t)" "NEXT(n)" "|" "DONE(d!)")
                          (sequence "WAITING(w@/!)" "HOLD(h@/!)" "|" "CANCELLED(c@/!)"))
      org-todo-keyword-faces '(
                               ("TODO" . (:foreground "#cc241d" :weight bold :underline t))
                               ("NEXT" . (:foreground "#fe8019" :weight bold))
                               ("DONE" . (:foreground "#83a598" :weight bold :strike-through t))
                               ("WAITING" . (:foreground "#d3869b" :weight bold :slant italic))
                               ("CANCELLED" . (:foreground "#83a598" :weight bold :strike-through t))
                               ))
)
#+end_src

** Enabling Org-Super-Agenda
Here is where I am setting up org-super agenda, which lets me make much more complex agenda views than I could with the regular org-mode agenda language.

#+begin_src emacs-lisp
(use-package! org-super-agenda
  :after org-agenda
  :init
  (require 'evil-org-agenda)
  :config
  (org-super-agenda-mode))
#+end_src

** Custom Agenda Commands
This is the section where I create custom agendas.
* Applications
** Mu4e
Mu4e is an excellent email program. I'm setting p my account here.
#+begin_src emacs-lisp
(add-to-list 'load-path "/usr/local/share/emacs/site-lisp/mu4e") ;; On Ubuntu
  (require 'mu4e);; on Ubuntu

  (require 'org-mu4e) ;; On Ubuntu
(setq mu4e-maildir "~/.local/share/mail")
(setq mu4e-contexts
	(list
	 ;; Primary personal account
	 (make-mu4e-context
	  :name "jhilker2"
	  :match-func
	    (lambda (msg)
	      (when msg
		(string-prefix-p "/jacob.hilker2" (mu4e-message-field msg :maildir))))
	  :vars '((user-mail-address . "jacob.hilker2@gmail.com")
		  (user-full-name    . "Jacob Hilker")
		  (smtpmail-smtp-server  . "smtp.gmail.com")
        (mu4e-get-mail-func "mbsync jacob.hilker2@gmail.com")
		  (smtpmail-smtp-service . 465)
		  (smtpmail-stream-type  . ssl)
		  (mu4e-drafts-folder  . "/jacob.hilker2@gmail.com/[Gmail]/Drafts")
		  (mu4e-sent-folder  . "/jacob.hilker2@gmail.com/[Gmail]/Sent Mail")
		  (mu4e-refile-folder  . "/jacob.hilker2@gmail.com/[Gmail]/All Mail")
		  (mu4e-trash-folder  . "/jacob.hilker2@gmail.com/[Gmail]/Trash")))))

#+end_src

** Elfeed
Elfeed is a rss reader for emacs.
#+begin_src emacs-lisp
(after! elfeed
(elfeed-goodies/setup)
(setq elfeed-search-filter "@1-week-ago +unread"
      elfeed-goodies/feed-source-column-width 30
      elfeed-goodies/tag-column-width 16)
)
#+end_src

* Minor Tweaks
** Relative Line Numbers
#+begin_src emacs-lisp
(setq display-line-numbers-type 'relative)
#+end_src

* Keybindings
** Spacemacs =M-x=
#+begin_src emacs-lisp
(map! :leader
      :desc "Open like spacemacs" "SPC" #'counsel-M-x)
#+end_src

** Org-Babel
#+begin_src emacs-lisp
(map! :leader
      :desc "Org babel tangle" "m B" #'org-babel-tangle)
#+end_src

** Open Programs
These are the keybindings I want to use for opening both agenda custom commands and for various programs.
#+begin_src emacs-lisp

(map! :leader
      (:prefix ("o" . "open")
       :desc "News reader" "n" #'elfeed))

#+end_src


.doom.d/config.el
(setq user-full-name "Jacob Hilker"
      user-mail-address "jacob.hilker2@gmail.com")

(setq doom-theme 'doom-gruvbox)

(setq doom-font (font-spec :family "Iosevka Nerd Font" :size 14)
      doom-variable-pitch-font (font-spec :family "Iosevka Aile" :size 14))

(after! doom-themes
  (setq
   doom-themes-enable-bold t
   doom-themes-enable-italic t))

(custom-set-faces!
  '(font-lock-comment-face :slant italic))

(after! org
  (set-face-attribute 'org-level-1 nil :foreground "#83a598")
  (set-face-attribute 'org-level-2 nil :foreground "#d3869b")
  (set-face-attribute 'org-level-3 nil :foreground "#fabd2f")
  (set-face-attribute 'org-level-4 nil :foreground "#fb4934")
  (set-face-attribute 'org-level-5 nil :foreground "#83a598")
  (set-face-attribute 'org-level-6 nil :foreground "#d3869b")
  (set-face-attribute 'org-level-7 nil :foreground "#fabd2f")
  (set-face-attribute 'org-level-8 nil :foreground "#fb4934"))

(after! org
(setq
      org-directory "~/Dropbox/org"
      org-agenda-files '("~/Dropbox/org/agenda.org" "~/Dropbox/org/projects.org" "~/Dropbox/org/contacts.org" "~/Dropbox/org/todo.org")
      org-todo-keywords '((sequence "TODO(t)" "NEXT(n)" "|" "DONE(d!)")
                          (sequence "WAITING(w@/!)" "HOLD(h@/!)" "|" "CANCELLED(c@/!)"))
      org-todo-keyword-faces '(
                               ("TODO" . (:foreground "#cc241d" :weight bold :underline t))
                               ("NEXT" . (:foreground "#fe8019" :weight bold))
                               ("DONE" . (:foreground "#83a598" :weight bold :strike-through t))
                               ("WAITING" . (:foreground "#d3869b" :weight bold :slant italic))
                               ("CANCELLED" . (:foreground "#83a598" :weight bold :strike-through t))
                               ))
)

(use-package! org-super-agenda
  :after org-agenda
  :init
  (require 'evil-org-agenda)
  :config
  (org-super-agenda-mode))

(add-to-list 'load-path "/usr/local/share/emacs/site-lisp/mu4e") ;; On Ubuntu
  (require 'mu4e);; on Ubuntu

  (require 'org-mu4e) ;; On Ubuntu
(setq mu4e-maildir "~/.local/share/mail")
(setq mu4e-contexts
	(list
	 ;; Primary personal account
	 (make-mu4e-context
	  :name "jhilker2"
	  :match-func
	    (lambda (msg)
	      (when msg
		(string-prefix-p "/jacob.hilker2" (mu4e-message-field msg :maildir))))
	  :vars '((user-mail-address . "jacob.hilker2@gmail.com")
		  (user-full-name    . "Jacob Hilker")
		  (smtpmail-smtp-server  . "smtp.gmail.com")
        (mu4e-get-mail-func "mbsync jacob.hilker2@gmail.com")
		  (smtpmail-smtp-service . 465)
		  (smtpmail-stream-type  . ssl)
		  (mu4e-drafts-folder  . "/jacob.hilker2@gmail.com/[Gmail]/Drafts")
		  (mu4e-sent-folder  . "/jacob.hilker2@gmail.com/[Gmail]/Sent Mail")
		  (mu4e-refile-folder  . "/jacob.hilker2@gmail.com/[Gmail]/All Mail")
		  (mu4e-trash-folder  . "/jacob.hilker2@gmail.com/[Gmail]/Trash")))))

(after! elfeed
(elfeed-goodies/setup)
(setq elfeed-search-filter "@1-week-ago +unread"
      elfeed-goodies/feed-source-column-width 30
      elfeed-goodies/tag-column-width 16)
)

(setq display-line-numbers-type 'relative)

(map! :leader
      :desc "Open like spacemacs" "SPC" #'counsel-M-x)

(map! :leader
      :desc "Org babel tangle" "m B" #'org-babel-tangle)

(map! :leader
      (:prefix ("o" . "open")
       :desc "News reader" "n" #'elfeed))
